renderdoc/api/replay/structured_data.h

```diff
@@ -676,10 +776,19 @@ inline SDObject *makeSDStruct(const char *name, const char *structtype)
 // concept of different width types like 32-bit vs 64-bit ints
 #if !defined(SWIG)
 
-#define SDOBJECT_MAKER(basetype, makeSDFunc)                      \
-  inline SDObject *makeSDObject(const char *name, basetype value) \
-  {                                                               \
-    return makeSDFunc(name, value);                               \
+#define SDOBJECT_MAKER(basetype, makeSDFunc)                                                \
+  inline SDObject *makeSDObject(const char *name, basetype value)                           \
+  {                                                                                         \
+    return makeSDFunc(name, value, NULL, NULL);                                             \
+  }                                                                                         \
+  inline SDObject *makeSDObject(const char *name, basetype value, const char *customString) \
+  {                                                                                         \
+    return makeSDFunc(name, value, customString, NULL);                                     \
+  }                                                                                         \
+  inline SDObject *makeSDObject(const char *name, basetype value, const char *customString, \
+                                const char *customTypeName)                                 \
+  {                                                                                         \
+    return makeSDFunc(name, value, customString, customTypeName);                           \
```

Comment:
Since this is C++ only it's fine to add extra optional parameters that will call `SetCustomString` / `SetTypeName` (see other comment) if you want. I'm not sure why you have all three variants listed separately though, instead of a single variant with default parameters, if there's no reason then it would be preferable to change it:

```cpp 
 #define SDOBJECT_MAKER(basetype, makeSDFunc)                                                       \
   inline SDObject *makeSDObject(const char *name, basetype value, const char *customString = NULL, \
                                 const char *customTypeName = NULL)                                 \
   {                                                                                                \
     return makeSDFunc(name, value, customString, customTypeName);                                  \
   }
```

Reply:
done